### ISSUE TITLE
Isolated Domain webpart not loading in Microsoft Team desktop client

### ISSUE THREAD

**Issue by vasanthtt** [SharePoint/sp-dev-docs#4669]
I know there are similar issues already reported but non are resolved and active now. 

I created a basic isolated webpart using SPfx 1.9.1 using latest Yeoman SharePoint Generator. Then, i followed below changes
1. updated api permission on package-solution.json
```
{
  "$schema": "https://developer.microsoft.com/json-schemas/spfx-build/package-solution.schema.json",
  "solution": {
    "name": "planisolated-wp-client-side-solution",
    "id": "cacd8aa4-7401-486d-9edd-b2fd6ba861f1",
    "version": "1.0.0.0",
    "includeClientSideAssets": true,
    "isDomainIsolated": true,
    "webApiPermissionRequests": [
      {
        "resource": "Microsoft Graph",
        "scope": "User.Read"
      },
      {
        "resource": "Windows Azure Active Directory",
        "scope": "User.Read"
      }
    ]
  },
  "paths": {
    "zippedPackage": "solution/planisolated-wp.sppkg"
  }
}
```
2. Added below code in webpart.ts file
```
import * as microsoftTeams from '@microsoft/teams-js';
...

private teamsContext: microsoftTeams.Context;

  protected onInit(): Promise<void> {
    return new Promise<void>((resolve, reject) => {
      if (this.context.microsoftTeams) {
        this.context.microsoftTeams.getContext(context => {
          this.teamsContext = context;
          resolve();
        });
      } else {
        resolve();
      }
    });
  }
```

3. Created package solution and uploaded in the tenant app catalog.
4. Approved both permissions in SharePoint admin central under API management page.
5. Added the webpart to site collection which Teams is already created.
6. Webpart working in site collection pages.
7. In Microsoft Teams windows client, the site collection teams and tried to add the webpart in tab. But failed to add(screenshot below)
![image](https://user-images.githubusercontent.com/724399/65401224-4fa4e780-dde4-11e9-9673-77b56ea4dbe7.png)

But I can able to do same in Microsoft Teams web client.

Please let me know if I missed anything. Thanks.

In fiddler, I see this

```
POST https://abc.sharepoint.com/sites/tt/_api/SP.OAuth.NativeClient/Authenticate HTTP/1.1
Host: abc.sharepoint.com
Connection: keep-alive
Content-Length: 0
Accept: */*
Accept-Encoding: gzip, deflate, br
Origin: https://abc-appcacd8aa47401486d9eddb2fd6ba861f1.sharepoint.com
Referer: https://abc-appcacd8aa47401486d9eddb2fd6ba861f1.sharepoint.com/sites/tt/_layouts/15/TeamsLogon.aspx?SPFX=true&dest=/sites/tt/_layouts/15/teamshostedapp.aspx%3FopenPropertyPane=true%26teams%26componentId=f4735292-1b9a-4fbe-9c73-aa0e8b5546de%26forceLocale=en-in
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Teams/1.2.00.26068 Chrome/66.0.3359.181 Electron/3.1.11 Safari/537.36
authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1N...


HTTP/1.1 401 Unauthorized
Server: Microsoft-IIS/10.0
Access-Control-Allow-Origin: *
Access-Control-Max-Age: 2592000
Access-Control-Expose-Headers: SPRequestGuid, X-MoveState, Timing-Allow-Origin, SPRequestDuration, Content-Length
x-ms-diagnostics: 3000003;reason="Invalid audience Uri 'https://abc-appcacd8aa47401486d9eddb2fd6ba861f1.sharepoint.com/'.";category="invalid_client"
SPRequestGuid: 5fea069f-3096-0000-336b-b2c4da8636b8
request-id: 5fea069f-3096-0000-336b-b2c4da8636b8
MS-CV: nwbqX5YwAAAza7LE2oY2uA.0
Strict-Transport-Security: max-age=31536000
X-FRAME-OPTIONS: SAMEORIGIN
SPRequestDuration: 7
SPIisLatency: 0
X-Powered-By: ASP.NET
MicrosoftSharePointTeamServices: 16.0.0.19311
X-Content-Type-Options: nosniff
X-MS-InvokeApp: 1; RequireReadOnly
P3P: CP="ALL IND DSP COR ADM CONo CUR CUSo IVAo IVDo PSA PSD TAI TELo OUR SAMo CNT COM INT NAV ONL PHY PRE PUR UNI"
WWW-Authenticate: Bearer realm="156eecee-ed1e-4497-acfe-407b7a5c924f",client_id="00000003-0000-0ff1-ce00-000000000000",trusted_issuers="00000001-0000-0000-c000-000000000000@*,D3776938-3DBA-481F-A652-4BEDFCAB7CD8@*,https://sts.windows.net/*/,00000003-0000-0ff1-ce00-000000000000@90140122-8516-11e1-8eff-49304924019b",authorization_uri="https://login.windows.net/common/oauth2/authorize"
Date: Mon, 23 Sep 2019 04:55:05 GMT
Content-Length: 123

{"error_description":"Exception of type 'Microsoft.IdentityModel.Tokens.AudienceUriValidationFailedException' was thrown."}
```

and
```
GET https://abc-appcacd8aa47401486d9eddb2fd6ba861f1.sharepoint.com/sites/tt/_layouts/15/teamshostedapp.aspx?openPropertyPane=true&teams&componentId=f4735292-1b9a-4fbe-9c73-aa0e8b5546de&forceLocale=en-in HTTP/1.1
Host: abc-appcacd8aa47401486d9eddb2fd6ba861f1.sharepoint.com
Connection: keep-alive
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Cookie: nSGt-CDA912FB41E26ACDB6B7AFC174F638D7563210C872E9170D=...
Referer: https://abc-appcacd8aa47401486d9eddb2fd6ba861f1.sharepoint.com/sites/tt/_layouts/15/TeamsLogon.aspx?SPFX=true&dest=/sites/tt/_layouts/15/teamshostedapp.aspx%3FopenPropertyPane=true%26teams%26componentId=f4735292-1b9a-4fbe-9c73-aa0e8b5546de%26forceLocale=en-in
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Teams/1.2.00.26068 Chrome/66.0.3359.181 Electron/3.1.11 Safari/537.36


HTTP/1.1 302 Found
Cache-Control: private
Content-Type: text/html; charset=utf-8
Location: /_forms/default.aspx?ReturnUrl=%2fsites%2ftt%2f_layouts%2f15%2fteamshostedapp.aspx%3fopenPropertyPane%3dtrue%26teams%26componentId%3df4735292-1b9a-4fbe-9c73-aa0e8b5546de%26forceLocale%3den-in&openPropertyPane=true&teams&componentId=f4735292-1b9a-4fbe-9c73-aa0e8b5546de&forceLocale=en-in
Server: Microsoft-IIS/10.0
X-SharePointHealthScore: 1
SPRequestGuid: 5fea069f-e09c-0000-32ed-f207c8885e3f
request-id: 5fea069f-e09c-0000-32ed-f207c8885e3f
MS-CV: nwbqX5zgAAAy7fIHyIhePw.0
Strict-Transport-Security: max-age=31536000
X-AspNet-Version: 4.0.30319
SPRequestDuration: 844
SPIisLatency: 2
X-Powered-By: ASP.NET
MicrosoftSharePointTeamServices: 16.0.0.19311
X-Content-Type-Options: nosniff
X-MS-InvokeApp: 1; RequireReadOnly
P3P: CP="ALL IND DSP COR ADM CONo CUR CUSo IVAo IVDo PSA PSD TAI TELo OUR SAMo CNT COM INT NAV ONL PHY PRE PUR UNI"
Date: Mon, 23 Sep 2019 04:55:05 GMT
Content-Length: 419

<html><head><title>Object moved</title></head><body>
<h2>Object moved to <a href="/_forms/default.aspx?ReturnUrl=%2fsites%2ftt%2f_layouts%2f15%2fteamshostedapp.aspx%3fopenPropertyPane%3dtrue%26teams%26componentId%3df4735292-1b9a-4fbe-9c73-aa0e8b5546de%26forceLocale%3den-in&amp;openPropertyPane=true&amp;teams&amp;componentId=f4735292-1b9a-4fbe-9c73-aa0e8b5546de&amp;forceLocale=en-in">here</a>.</h2>
</body></html>
```

**Comment by vasanthtt** [comment]
@andrewconnell Thanks for the response. 
My sample code [here](https://github.com/vasanthtt/SPFx/tree/master/simpleIsolatedWP)

**Comment by vasanthtt** [comment]
@lucabandMSFT  @andrewconnell When will we have a fix for loading Isolated webpart in rich clients. Thanks.

**Comment by muradhaider5** [comment]
Yeah still waiting.. actually facing two problems with Isolated Webpart:
1. (The ever-green issue) SPFx webpart containing graph API calls is not working in Teams Desktop client. Even though it's working in Teams web client and SharePoint site page.
2. For both desktop and web client, unable to get Team's context (using the same code in onInit() as written in issue description) in Webpart code to access different properties e.g. teamId, groupID etc, which was not the case with non-isolated Webpart.
Have tried almost all the workarounds that are mentioned in related issues: 
a. Manually updating the SharePoint Online Client Web Application Principle manifest (although it had 'preAuthorizedApplications' configured properly already).
b.  Removing all entries in the 'PasswordCredentials' section and leaving empty array there and visiting API management page afterwards.
Please break the silence.

**Comment by andrewconnell** [comment]
@lucabandMSFT is this the issue that was getting fixed in the Teams November client update announced at Ignite? That was my understanding from your session, and that there was nothing developers needed to do. It was an SPO service + Teams client update issue, not a SPFx version issue... correct?

**Comment by lucabandMSFT** [comment]
(+ @lahuey just for FYI) No, not yet. We are actively working on it and we are really hoping to be able to unlock the capability in February. That's not lack of commitment from our side but we want to do the right thing from all the aspects (performance, security, etc..).

**Comment by loflet** [comment]
Is there any update on this issue?

**Comment by DominikS90** [comment]
Any Updates with this issue?

**Comment by karthikramasubramanian** [comment]
@andrewconnell  has there been any updates\fixes\workarounds on the issue #6708?  It has been marked as closed.  But I was not able to find any resolution or workarounds in discussions for that issue..  
I updated the webpart to be domain isolated and since then these webparts are not working in the tabs of my teams app.  
Anyone know how I should update the manifest.json of my teams app so that these isolated webparts can be loaded.

**Comment by patmill** [comment]
Sorry for the very long delay - incorrect metadata hid this from us.  This is currently an unsupported scenario (isolated teams components).  It is on our backlog, but there is no explicit ETA.
